lib/Sema/TypeCheckType.cpp

```diff
@@ -4579,8 +4579,15 @@ NeverNullType TypeResolver::resolvePackElement(PackElementTypeRepr *repr,
     return ErrorType::get(ctx);
 
   if (!packReference->isParameterPack()) {
-    ctx.Diags.diagnose(repr->getLoc(), diag::each_non_pack,
-                       packReference);
+    auto diag =
+        ctx.Diags.diagnose(repr->getLoc(), diag::each_non_pack, packReference);
+    if (auto *packIdent = dyn_cast<IdentTypeRepr>(repr->getPackType())) {
+      if (auto *packIdentBinding = packIdent->getBoundDecl()) {
```

Comment:
Maybe `getNameLoc` is good enough here instead of relying on declaration to be there?

Reply:
Or is this repr from `repeat each T` argument from your example?

Reply:
Yes `repr` is the `repeat each T` and I am applying the fixit to the `T` back in `<T>`.
If I try using `packIdent->getNameLoc().getBaseNameLoc()` for the fixit location, then the fixit is applied to the T in `repeat each T` (leading to `repeat each each T`) rather than the T in `<T>`.

Reply:
I see, makes sense!